pkgs/by-name/to/tokyonight-gtk-theme/package.nix

```
@@ -45,7 +81,15 @@ stdenvNoCC.mkDerivation {
     runHook preInstall
     mkdir -p $out/share/themes
     cd themes
-    ./install.sh -n Tokyonight -c ${lib.concatStringsSep " " (if colorVariants != [] then colorVariants else colorVariantList)} --tweaks macos -d "$out/share/themes"
+    ./install.sh -n Tokyonight \
+    ${lib.optionalString (colorVariants != []) "-c " + toString colorVariants} \
+    ${lib.optionalString (sizeVariants != []) "-s " + toString sizeVariants} \
+    ${lib.optionalString (themeVariants != []) "-t " + toString themeVariants} \
+    ${lib.optionalString (tweakVersions != []) "--tweaks " + toString tweakVersions} \
+    -d "$out/share/themes"
+    cd ../icons
+    ${lib.optionalString (iconVariants != []) "mkdir -p $out/share/icons"}
+    ${lib.optionalString (iconVariants != []) "cp -a Tokyonight-" + toString iconVariants + " $out/share/icons/"}
```

```suggestion
    ${lib.optionalString (iconVariants != []) "cp -a " + toString (map (v: "Tokyonight-${v}" iconVariants)) + " $out/share/icons/"}
```

Otherwise this fails with multiple `iconVariants`.